collect_app/src/main/java/org/odk/collect/android/instancemanagement/InstanceListItemView.kt

```diff
@@ -0,0 +1,122 @@
+        setUpSubtext(view, instance, context)
+
+        val chip = view.findViewById<TextView>(R.id.chip)
+        if (chip != null) {
+            if (instance.status == Instance.STATUS_INVALID || instance.status == Instance.STATUS_INCOMPLETE) {
+                chip.visibility = View.VISIBLE
+            }
+        }
+
+        if (shouldCheckDisabled) {
+            var formExists = false
+            var isFormEncrypted = false
+            val formId = instance.formId
+            val formVersion = instance.formVersion
+            val form = FormsRepositoryProvider(context.applicationContext).get()
+                .getLatestByFormIdAndVersion(formId, formVersion)
+
+            if (form != null) {
+                val base64RSAPublicKey = form.basE64RSAPublicKey
+                formExists = true
+                isFormEncrypted = base64RSAPublicKey != null
+            }
+
+            val date = instance.deletedDate
+            if (date != null || !formExists || isFormEncrypted) {
+                val disabledMessage = if (date != null) {
+                    try {
+                        val deletedTime: String = context.getString(string.deleted_on_date_at_time)
+                        SimpleDateFormat(deletedTime, Locale.getDefault()).format(Date(date))
+                    } catch (e: IllegalArgumentException) {
+                        Timber.e(e)
+                        context.getString(string.submission_deleted)
+                    }
+                } else if (!formExists) {
+                    context.getString(string.deleted_form)
+                } else {
+                    context.getString(string.encrypted_form)
+                }
+
+                setDisabled(view, disabledMessage)
+            } else {
+                setEnabled(view)
+            }
+        }
+    }
+
+    private fun setEnabled(view: View) {
+        val formTitle = view.findViewById<TextView>(R.id.form_title)
+        val formSubtitle = view.findViewById<TextView>(R.id.form_subtitle)
+        val disabledCause = view.findViewById<TextView>(R.id.form_subtitle2)
+        val imageView = view.findViewById<ImageView>(R.id.image)
+        view.isEnabled = true
+        disabledCause.visibility = View.GONE
+        formTitle.alpha = 1f
+        formSubtitle.alpha = 1f
+        disabledCause.alpha = 1f
+        imageView.alpha = 1f
+    }
+
+    private fun setDisabled(view: View, disabledMessage: String) {
+        val formTitle = view.findViewById<TextView>(R.id.form_title)
+        val formSubtitle = view.findViewById<TextView>(R.id.form_subtitle)
+        val disabledCause = view.findViewById<TextView>(R.id.form_subtitle2)
+        val imageView = view.findViewById<ImageView>(R.id.image)
+        view.isEnabled = false
+        disabledCause.visibility = View.VISIBLE
+        disabledCause.text = disabledMessage
+
+        // Material design "disabled" opacity is 38%.
+        formTitle.alpha = 0.38f
```

Comment:
I think it would be good to factor out this constant.

Reply:
I think this might not even be correct anymore. I'll pull it out as a constant for this file, but don't want to get distracted by restyling the whole list item right now.